### DISCUSSION TITLE
❓ [Question] How to set three input tensor shape in input_shape?

### DISCUSSION THREAD

**Discussion by wxyhv** [pytorch/TensorRT#506] [Q&A] [answered]
## ❓ Question

I have three input tensor:src_tokens, dummy_embeded_x, dummy_encoder_embedding
In this case, I don't konw how to set input_shape in compile_settings's "input_shape"
Who can help me? Thank you!

`encoder_out = model.forward_encoder([src_tokens, dummy_embeded_x, dummy_encoder_embedding])`
`...`
`script_encoder = torch.jit.script(encoder)...`
`compile_settings = {
            "input_shapes": [[2, 16]],
            "op_precision": torch.float32
        }`

**Comment by narendasan** [comment, marked as answer]
You set the input shapes as a list of lists or a list of dicts if you are using dynamic shape.

So for your 3 inputs you would do something like this:

```py
"input_shapes": [[2,16], [1,10], [1, 15]]
```

or something similar where the entries correspond to src_tokens, dummy_embeded_x, dummy_encoder_embedding in the order they will appear in the function signature for your forward i.e. in this case maybe something like `forward(src_tokens, dummy_embeded_x, dummy_encoder_embedding)`.

To do dynamic shape it would look like this: 


```py
"input_shapes": [
  {
    "min": [2, 16],
    "opt": [2, 20],
    "max: [2, 25],
   }, {
    "min": [1, 10],
    "opt": [1, 20],
    "max: [1, 25],
   }, {
    "min": [1, 20],
    "opt": [1, 25],
    "max: [1, 35],
   }
]
```
